src/operators/mpoham.jl

```diff
@@ -83,9 +83,18 @@ function deduce_pspaces(opps::SumOfLocalOperators)
     end
 
     non_deduced = map(ismissing, pspaces)
-    any(non_deduced) &&
-        error("cannot automatically deduce physical spaces at $(findall(non_deduced))")
 
+    if any(non_deduced) # Some spaces were not defined / not able to be deduced
+        if allequal(filter(!ismissing, pspaces)) # all non-missing spaces are equal
```

Comment:
It probably doesn't matter too much, but this does look like you are recomputing the same thing quite a few times